packages/app-desktop/app.ts

```diff
@@ -405,12 +405,17 @@ class Application extends BaseApplication {
 	}
 
 	private setupAutoUpdaterService() {
+		// since the remote process doesn't stop running after app is closed, we need to initialize the service, even if its flag is set to false, or else it will throw an error.
+		bridge().electronApp().initializeAutoUpdaterService(
+			Logger.create('AutoUpdaterService'),
+			Setting.value('env') === 'dev',
+			Setting.value('autoUpdate.includePreReleases'),
+		);
```

Comment:
Likewise we can't have this running if the user didn't enable the feature. That's the purpose of this flag, to enable it in a controlled way. I don't really understand the logic here, why we need to run this if the flag is not on? If the user needs to restart the app after enabling the flag, we just document it. The flag is a temporary, advanced feature anyway

Reply:
![image](https://github.com/user-attachments/assets/a074937d-4d99-4682-874f-bf510b1d9005)

In the explanation below, by saying "default flag" I refer to the flag previously used for auto updates, together with the existing logic from before. By saying "implemented flag", I refer to the flag I added for my service.

Explanation: let's say the user has the default flag set to true and the implemented flag set to false. If he/she goes to Options to set the implemented flag to true, and then goes back into the application to manually check for an update without closing or quitting the app, the above error will be shown, because the service was never initialized in the first place (since the implemented flag was initially false).

I even tried to add this check so we could avoid this situation:
`if (typeof this.updaterService_ != 'undefined' && this.updaterService_ && this.updaterService_?.checkForUpdates) `
Unfortunately it still throws the error.

Reply:
Not sure about the technical details, but it should be possible to turn on your feature flag and it only uses your new updater service, or turn it off and it only uses the old method. The fact that something is not working when one of the flag is on or off means some logic is missing somewhere